app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -124,4 +133,124 @@ public Void then(Task<JSONObject> task) throws Exception {
 
         return networkRequest.runNetworkRequest().continueWith(continuation);
     }
+
+    public static Task<JSONObject> addGroceryPhoto(final String groceryId, byte[] data) {
+
+
+        final Task<JSONObject>.TaskCompletionSource taskCompletionSource = Task.create();
+
+        Continuation<JSONObject, Void> addPhotoNameToGroceryPhoto = new Continuation<JSONObject, Void>() {
+            @Override
+            public Void then(Task<JSONObject> task) throws Exception {
+                if (task.isFaulted()) {
+                    Exception exception = task.getError();
+                    Log.e(TAG, "Error in uploadPhoto", exception);
+                    throw exception;
+                }
+
+                String photoName = task.getResult().optString("name");
+                if (TextUtils.isEmpty(photoName)) {
+                    throw new InvalidResponseException("Empty response");
+                }
+
+                final UrlTemplate template = UrlTemplateCreator.addGroceryPhoto(groceryId, photoName);
+
+                Continuation<JSONObject, Void> addingGroceryPhoto = new Continuation<JSONObject, Void>() {
+                    @Override
+                    public Void then(Task<JSONObject> task) throws Exception{
+                        if (task.isFaulted()) {
+                            Exception exception = task.getError();
+                            Log.e(TAG, "Error in addingGroceryPhoto", exception);
+                            throw exception;
+                        }
+
+                        try {
+                            Log.d(TAG, "adding to GroceryPhoto:" + task.getResult().toString());
+                            String createdAt = task.getResult().getString("createdAt");
+                            if (TextUtils.isEmpty(createdAt)) {
+                                throw new InvalidResponseException("Incorrect adding to GroceryPhoto response");
+                            }
+                        } catch (JSONException e) {
+                            Log.e(TAG, "Error parsing adding grocery photo response", e);
+                        }
+                        return null;
+                    }
+                };
+                new NetworkRequest(template, taskCompletionSource).runNetworkRequest().continueWith(addingGroceryPhoto);
+
+                return null;
+            }
+        };
+
+        String photoName = Utilities.dateToString(new Date(), MyApplication.getContext()
+                .getString(R.string.photo_date_format_string)) + ".jpg";
+        SyncPhoto.uploadPhoto(photoName, data).continueWith(addPhotoNameToGroceryPhoto);
+
+        return taskCompletionSource.getTask();
+    }
+
+    public static void getAllGroceryPhotos() {
```

Comment:
Whenever you get anything from Parse, you need to filter by groupId at a minimum.